onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -503,15 +536,57 @@ void UpdateCudaProviderOptions(InferenceSession* sess, onnxruntime::CudaProvider
     LOGS(*(sess->GetLogger()), INFO) << "cuda arean extend strategy is set to " << it->second;
   }
 }
+
+AllocatorPtr GetCudaAllocator(OrtDevice::DeviceId id) {
+  // Current approach is not thread-safe, but there are some bigger infra pieces to put together in order to make
```

Comment:
Not thread safe in python. But realistically, we don't need support for it in Python. We need to support something like the PerThreadContext in the CUDA EP to enable support. We can re-visit if it is needed.